l3sys-query.sty

```diff
@@ -0,0 +1,72 @@
+
+\ProvidesPackage{l3sys-query}[2024-03-08 v0.01 LaTeX2e interface for l3sys file queries]
+
+% Copyright 2024  The LaTeX Project
+% License: LPPL 1.3c
+
+% Defines:
+
+% \QueryPWD {<result>}
+% defines the command passed in as #1 to be the current working directory
+
+% \QueryFiles[<options>]{<args>}{function}
```

Comment:
Again not sure about the name but I don't have an immediate suggestion.

Also I wonder if you want the case that an empty list is returned handled by a sparate argument perhaps (just a thought)

Reply:
could add two more and execute <True code> <over list> <false code>   or perhaps 

or three more:

`\QueryFiles[<options>]{<args>}{pre}{function}{post}{empty}`



I was aiming for "something simple" here though with the intention of documenting if you want to do more drop down to L3 layer where you just get returned a `seq` variable and you can do whatever  you want with that.

Reply:
I added a version `\QueryFilesTF[<options>]{<args>}{function}{pre code}{empty code}` that prefixes the non empty list or slips the mapping and uses the empty code if the underlying seq is emty.  I  think that covers the main use cases.